### ISSUE TITLE
Chrome Version 121 -Could not instantiate class org.openqa.selenium.chrome.ChromeDriver

### ISSUE THREAD

**Issue by tarun04sharma** [serenity-bdd/serenity-core#3408]
Could not instantiate new WebDriver instance of type class org.openqa.selenium.chrome.ChromeDriver (Could not start a new session. Response code 500. Message: session not created: This version of ChromeDriver only supports Chrome version 114
Current browser version is 121.0.6167.184 with binary path /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
Build info: version: '4.1.2' .

had a similar issue in August with Chrome 116 update , is there any solution for this now .

I am using the below versions , 
<serenity.version>3.2.1</serenity.version>
 <groupId>io.github.bonigarcia</groupId>
            <artifactId>webdrivermanager</artifactId>
            <version>5.5.0</version>
 
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-chrome-driver</artifactId>
            <version>4.1.2</version>
Similar to the issue https://github.com/serenity-bdd/serenity-core/issues/3227 , any fix for this ?

@wakaleo

**Comment by mbarua1** [comment]
I am also having same issue and after upgrading serenity version from 3.6 to 4.0 now stuck in ` TestRunner » AbstractMethod Receiver class io.cucumber.junit.JUnitCucumberOptionsProvider$JunitCucumberOptions does not define or inherit an implementation of the resolved method 'abstract java.lang.Class uuidGenerator()' of interface io.cucumber.core.options.CucumberOptionsAnnotationParser$CucumberOption`s

[INFO] 
`

**Comment by wakaleo** [comment]
> I am also having same issue and after upgrading serenity version from 3.6 to 4.0 now stuck in ` TestRunner » AbstractMethod Receiver class io.cucumber.junit.JUnitCucumberOptionsProvider$JunitCucumberOptions does not define or inherit an implementation of the resolved method 'abstract java.lang.Class uuidGenerator()' of interface io.cucumber.core.options.CucumberOptionsAnnotationParser$CucumberOption`s
> 
> [INFO] `

That is a different issue, most likely a dependency conflict.

**Comment by mbarua1** [comment]
Thanks John @wakaleo . Okay and not able to identify the dependency conflict. Can you help me please?

**Comment by wakaleo** [comment]
> Could not instantiate new WebDriver instance of type class org.openqa.selenium.chrome.ChromeDriver (Could not start a new session. Response code 500. Message: session not created: This version of ChromeDriver only supports Chrome version 114 Current browser version is 121.0.6167.184 with binary path /Applications/Google Chrome.app/Contents/MacOS/Google Chrome Build info: version: '4.1.2' .
> 
> had a similar issue in August with Chrome 116 update , is there any solution for this now .
> 
> I am using the below versions , <serenity.version>3.2.1</serenity.version> io.github.bonigarcia webdrivermanager 5.5.0
> 
> ```
>         <groupId>org.seleniumhq.selenium</groupId>
>         <artifactId>selenium-chrome-driver</artifactId>
>         <version>4.1.2</version>
> ```
> 
> Similar to the issue #3227 , any fix for this ?
> 
> @wakaleo

This isn't a Serenity issue directly, it is possibly related to old dependencies or drivers in your project or on your system path. See https://github.com/serenity-bdd/serenity-cucumber-starter for an example of the correct dependency configuration.

**Comment by mbarua1** [comment]
Thanks John @wakaleo . above Pom helped to resolved Chrome driver issue. However I am now having below issue. Injectors.getInjector method not supporting in new serenity version. Is there any way to resolve this situation.

[Injectors.getInjector() method is not supported in Serenity version 4.0.28](https://stackoverflow.com/questions/77691007/injectors-getinjector-method-is-not-supported-in-serenity-version-4-0-28)

**Comment by wakaleo** [comment]
> Thanks John @wakaleo . above Pom helped to resolved Chrome driver issue. However I am now having below issue. Injectors.getInjector method not supporting in new serenity version. Is there any way to resolve this situation.
> 
> [Injectors.getInjector() method is not supported in Serenity version 4.0.28](https://stackoverflow.com/questions/77691007/injectors-getinjector-method-is-not-supported-in-serenity-version-4-0-28)

Read the release notes for 4.0.1

**Comment by mbarua1** [comment]
Thanks John @wakaleo . It worked and was able to reconfigure environment variable. Now another issue arises Parallel execution is not working. I had parallel execution from 3.2, 3.6 and with same configuration it is not working for 4.1.0. I had max pool size and parallal for 4 thread. Initially it is opening 4 thread but from second go was opening one by one browser which increased my total execution time from 28min to 56min. Please advise me where to find info from this issue. I do not see any release note for parallel in 4.0.1

**Comment by wakaleo** [comment]
Parallel execution with JUnit 5 is different to JUnit 4 - have a look at the JUnit 5 docs to see how it works.

**Comment by mbarua1** [comment]
I am using Junit5

**Comment by wakaleo** [comment]
See https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-parallel-execution

**Comment by mbarua1** [comment]
@wakaleo John, I tried all possible solution according to doc you provided still not working. So What happened to my case, I was running parallel for 3 thread and running scenarion outline for 4 test users, i am expecting to open 3 browser in first initiative and one more browser in 2nd attempt to complete same steps with 4 users in parallel execution 3, rather I see 3 browser opened in 1st attempt (which is expected), then more from 2nd, 3rd initiative it is opening single browser and keep continuing for couple times. below is my Junit-Platform-properties file
`cucumber.execution.parallel.enabled=true
cucumber.execution.parallel.config.strategy=fixed
cucumber.execution.parallel.config.fixed.parallelism=3
cucumber.plugin=io.cucumber.core.plugin.SerenityReporterParallel
cucumber.execution.parallel.config.fixed.max-pool-size=3

And POM

> perties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <serenity.version>4.1.0</serenity.version>
        <junit-vintage-engine.version>5.10.1</junit-vintage-engine.version>
        <junit-platform-suite.version>1.10.1</junit-platform-suite.version>
        <cucumber-junit-platform-engine.version>7.15.0</cucumber-junit-platform-engine.version>
        <encoding>UTF-8</encoding>
        <tags></tags>
       
    </properties>
    <dependencies>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-core</artifactId>
            <version>${serenity.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-junit5</artifactId>
            <version>${serenity.version}</version>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-cucumber</artifactId>
            <version>${serenity.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-screenplay</artifactId>
            <version>${serenity.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-screenplay-webdriver</artifactId>
            <version>${serenity.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-ensure</artifactId>
            <version>${serenity.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.junit.platform</groupId>
            <artifactId>junit-platform-launcher</artifactId>
            <version>${junit-platform-suite.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-junit-platform-engine</artifactId>
            <version>${cucumber-junit-platform-engine.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.junit.platform</groupId>
            <artifactId>junit-platform-suite</artifactId>
            <version>${junit-platform-suite.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.junit.jupiter</groupId>
            <artifactId>junit-jupiter-engine</artifactId>
            <version>${junit-vintage-engine.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.junit.vintage</groupId>
            <artifactId>junit-vintage-engine</artifactId>
            <version>${junit-vintage-engine.version}</version>
            <scope>test</scope>
        </dependency>
      
         <dependency>
         	<groupId>jline</groupId>
         	<artifactId>jline</artifactId>
         	<version>2.11</version>	
      	</dependency>
      	<dependency>
    		<groupId>io.rest-assured</groupId>
    		<artifactId>rest-assured</artifactId>
    		<version>5.3.0</version>
    		<scope>test</scope>
		</dependency>
	<dependency>
    		<groupId>com.github.javafaker</groupId>
    		<artifactId>javafaker</artifactId>
    		<version>1.0.2</version>
		</dependency>
		
		
        <dependency>
            <groupId>org.assertj</groupId>
            <artifactId>assertj-core</artifactId>
            <version>3.23.1</version>
            <scope>test</scope>
        </dependency>
    </dependencies>
    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
               
                <version>3.2.1</version>
                <configuration>
                    <skip>true</skip>
                </configuration>
            </plugin>
            <plugin>
                <artifactId>maven-failsafe-plugin</artifactId>
               
                <version>3.2.1</version>
                <configuration>
                    <includes>
                        <include>**/*Test.java</include>
                        <include>**/Test*.java</include>
                        <include>**/*TestSuite.java</include>
                        <include>**/When*.java</include>
                    </includes>
                  
                    <parallel>classes</parallel>
                    <parallel>methods</parallel>
                   
                    <useUnlimitedThreads>true</useUnlimitedThreads> 
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>integration-test</goal>
                            <goal>verify</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.1</version>
                <configuration>
                    <source>1.8</source>
                    <target>1.8</target>
                </configuration>
            </plugin>
            <plugin>
                <groupId>net.serenity-bdd.maven.plugins</groupId>
                <artifactId>serenity-maven-plugin</artifactId>
                <version>${serenity.version}</version>
                <configuration>
                    <tags>${tags}</tags>
                    <reports>single-page-html</reports>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>net.serenity-bdd</groupId>
                        <artifactId>serenity-single-page-report</artifactId>
                        <version>${serenity.version}</version>
                    </dependency>
                </dependencies>
                <executions>
                    <execution>
                        <id>serenity-reports</id>
                        <phase>post-integration-test</phase>
                        <goals>
                            <goal>aggregate</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</project>

**Comment by mbarua1** [comment]
@tarun04sharma - Is parallel working for you?

**Comment by tarun04sharma** [comment]
@mbarua1  I have not yet implemented and tried parallel testing

**Comment by wakaleo** [comment]
@mbarua1 see  https://github.com/serenity-bdd/screenplay-pattern-todomvc.git for an example of how to configure parallel execution with JUnit 5 and Cucumber in Serenity.

**Comment by mbarua1** [comment]
@wakaleo - Thanks John. Doing it now

**Comment by mbarua1** [comment]
John, I tried all options enlisted here with max pool size 4 and 3 test cases(just scenario, not scenario outline), So initially 3 browser opened and all 3 scenario (from same feature file) ran, then i am expecting to finish test execution and generate report. However it is not and again one by one browser was opening and one by one test was running. So total six browser opened (3 in first initiative and one by one). And report says 3 test ran which is weird, I am not understanding which causing this issue. With same set up (junit-platform file and with 4 max pool size )just with serenity 3.6 , I ws running same sets(30 test case) of smoke suite for last 8 month within 25 min and now it is taking 55 just for double run. Initially running as parallel and after a few min open one by one browser. Please suggest me how I can resolve this issue. @wakaleo

**Comment by wakaleo** [comment]
I would need to look at the project setup. Sometimes JUnit 5 can be tricky to configure. If you do need project-specific support we can do this as part of a commercial support contract.

**Comment by mbarua1** [comment]
alright, How to do it and what will be procedure for that?
@wakaleo

**Comment by wakaleo** [comment]
> alright, How to do it and what will be procedure for that? @wakaleo

See https://www.serenitydojo.academy/serenity-bdd-support-packages
 for all the options
